lib/email_tracking_retriever.py

```diff
@@ -111,37 +136,50 @@ def get_tracking_number_from_email(self,
     """
     pass
 
+
   @abstractmethod
   def get_subject_searches(self) -> Any:
     pass
 
+
   @abstractmethod
   def get_merchant(self) -> str:
     pass
 
+
   @abstractmethod
   def get_items_from_email(self, data) -> Any:
     pass
 
+
   @abstractmethod
   def get_delivery_date_from_email(self, data) -> Any:
     pass
 
+
   def get_date_from_msg(self, data) -> str:
     msg = email.message_from_string(str(data[0][1], 'utf-8'))
     msg_date = msg['Date']
     return datetime.datetime.strptime(
         msg_date, '%a, %d %b %Y %H:%M:%S %z').strftime('%Y-%m-%d')
 
+
   def get_to_address(self, data) -> str:
     msg = email.message_from_string(str(data[0][1], 'utf-8'))
     return str(msg['To']).replace('<', '').replace('>', '')
 
+
   @retry(
       stop=stop_after_attempt(3),
       wait=wait_exponential(multiplier=1, min=2, max=16),
       reraise=True)
-  def get_tracking(self, email_id, mail) -> Tracking:
+  def get_tracking(self, email_id, mail) -> Tuple[bool, Tracking]:
```

Comment:
someone has been looking at golang, eh?

Reply:
I haven't but this particular pattern makes so much sense that I'm essentially inventing it independently.

Also that's not true, this is a common pattern in older languages too, like C.

Python makes it easy to do with multiple return values though.